app/controllers/api/comments_controller.rb

```diff
@@ -0,0 +1,25 @@
+class Api::CommentsController < ApplicationController
+  skip_before_action :authenticate_user!
+  skip_before_action :verify_authenticity_token
+  def index
+    @comments = Comment.all
+    render json: @comments
+  end
+
+  def create
+    @post = Post.find(params[:post_id])
+    @comment = @post.comments.create(text: params[:text], user_id: params[:user_id])
+
+    if @comment.save
+      render json: @comment, status: :created
+    else
+      render json: { errors: @comment.errors.full_messages }, status: :unprocessable_entity
+    end
+  end
+
+  private
+
+  def comment_params
+    params.require(:comment).permit(:text, :user_id, :post_id)
+  end
```

Comment:
- [ ] I'm still not able to add a new comment to a post, but this time I'm getting a message that the text attribute can't be blank (see image). Seen the effort you put in this project, I try to troubleshoot your code and I come up with the following suggestion:
  - At line 11, kindly replace the `params[:text]` by `comment_params[:text]` to get the **text** value from the the information send by user. Your `create` method will look like this :point_down: after the changes is made.
```rb
  def create
    @post = Post.find(params[:post_id])
    @comment = @post.comments.create(text: comment_params[:text], user_id: params[:user_id])

    if @comment.save
      render json: @comment, status: :created
    else
      render json: { errors: @comment.errors.full_messages }, status: :unprocessable_entity
    end
  end
```
  - Please remove the `:user_id` and `:post_id` params from the `comment_params` method because there are already available through the GET(we can get them through the URL  `params[:user_id]` and `params[:post_id]`). Your `comment_params` method will look like this :point_down: after the changes
   ```rb
      private

      def comment_params
        params.require(:comment).permit(:text)
      end
   ```
![Screenshot from 2023-12-15 18-34-18](https://github.com/Megagig/Blog-App/assets/91142432/590ba266-fe9e-491e-9f04-5b434e4f7430)

Reply:
@LionRouge1 Everything works perfectly from my end. Please your connection properly.Below is the screenshot of work working perfectly on my computer.
![Uploading CreatingPost.PNG…]()

Reply:
Check this second screenshot. The code is working even without a parameter passed
<img width="868" alt="parameter" src="https://github.com/Megagig/Blog-App/assets/108594742/209c978a-b1b6-4bdc-8b41-7de728ec9e6b">